chore(StatusChatList): Keep sync with underlying model after reordering

micieslak opened the pull request:
### Checklist

After reordering chats or chat categories the order of elements in the list should reflect order in the underlying model. Reorders done via DelegateModel when dragging reordered item should be revoked to avoid improper sorting.

required by: status-im/status-desktop#6597, status-im/status-desktop#6722

affected:
StatusChatList, StatusChatListAndCategories

- [x] follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
  - the scope should be the component's name e.g: `feat(StatusListItem): ... `
  - when adding new components, the scope is the module e.g: `feat(StatusQ.Controls): ...`
- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?
- [ ] is this a breaking change?
    - [ ] use the dedicated `BREAKING CHANGE` commit message section
    - [ ] resolve breaking changes in [status-desktop](https://github.com/status-im/status-desktop)
        - [ ] (pre-merge) adapt code to breaking changes
        - [ ] (post-merge) update StatusQ submodule pointer
- [x] test changes in [status-desktop](https://github.com/status-im/status-desktop)

MishkaRogachev commented on src/StatusQ/Components/StatusChatList.qml:
```diff
@@ -33,6 +33,27 @@ Column {
     }
 
+    QtObject {
+        id: internal
```
we usually name this internal QObject as`d`

MishkaRogachev commented on src/StatusQ/Components/StatusChatListAndCategories.qml:
```diff
@@ -91,13 +91,30 @@ Item {
                 }
 
+                // undo any reorders done in a DelegateModel
+                function resetDelegateModel() {
```
There are two quite-similar functions, can we move it to js helper?

micieslak commented:
Thanks @MishkaRogachev for the review. I fixed what you mentioned but also changed the approach because I noticed some issues with the previous one when new items are added to the list. Now it seems to work fine and is slightly more "standard".

MishkaRogachev reviewed (approved):
Looks solid!

caybro reviewed (approved):
Changes look good

I just don't think this DelegateModel is a viable choice for the future :) So much imperative code
